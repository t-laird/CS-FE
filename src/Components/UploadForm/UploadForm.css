.UploadForm {
  width: 600px;
  margin: 0px auto;
  display: flex; }
  .UploadForm button {
    background: white;
    border: 1px solid black;
    transition: 0.3s;
    font-size: 20px;
    border-radius: 3px; }
    .UploadForm button:hover {
      color: white;
      background: #02A364; }
  .UploadForm form {
    display: grid;
    grid-template-columns: 1fr 100px;
    margin: 30px auto; }
    .UploadForm form input {
      font-size: 20px;
      color: white;
      height: 100%; }
    .UploadForm form span {
      min-height: 30px;
      padding-top: 8px;
      display: flex;
      justify-content: center;
      grid-column: span 2; }
      .UploadForm form span.neg {
        color: red; }
      .UploadForm form span.pos {
        color: #34D27C; }
