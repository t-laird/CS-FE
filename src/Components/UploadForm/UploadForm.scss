.UploadForm {
  width: 600px;
  margin: 0px auto;
  display: flex;

  button {
    background: white;
    border: 1px solid black;
    transition: 0.3s;
    font-size: 20px;
    border-radius: 3px;
    &:hover {
      color: white;
      background: #02A364;
    }
  }

  form {
    display: grid;
    grid-template-columns: 1fr 100px;
    margin: 30px auto;
    input {
      font-size: 20px;
      color: white;
      height: 100%;
    }
  
    span {
      min-height: 30px;
      padding-top: 8px;
      display: flex;
      justify-content: center;
      grid-column: span 2;
      &.neg {
        color: red;
      }
      &.pos {
        color: #34D27C;
      }
    }
  }


}