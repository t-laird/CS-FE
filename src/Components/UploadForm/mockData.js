module.exports = {
  "search": {
    "startDate": "2016-06-04",
    "endDate": "2016-06-06"
  },
  "campsites": [
    {
      "id": 1,
      "name": "Grizzly Adams Adventure Cabin"
    },
    {
      "id": 2,
      "name": "Lewis and Clark Camp Spot"
    },
    {
      "id": 3,
      "name": "Jonny Appleseed Log Cabin"
    },
    {
      "id": 4,
      "name": "Davey Crockett Camphouse"
    },
    {
      "id": 5,
      "name": "Daniel Boone Bungalo"
    }
  ],
  "reservations": [
    {"campsiteId": 1, "startDate": "2016-06-01", "endDate": "2016-06-03"},
    {"campsiteId": 1, "startDate": "2016-06-08", "endDate": "2016-06-10"},
    {"campsiteId": 2, "startDate": "2016-06-01", "endDate": "2016-06-01"},
    {"campsiteId": 2, "startDate": "2016-06-07", "endDate": "2016-06-09"},
    {"campsiteId": 3, "startDate": "2016-06-01", "endDate": "2016-06-02"},
    {"campsiteId": 3, "startDate": "2016-06-08", "endDate": "2016-06-09"},
    {"campsiteId": 4, "startDate": "2016-06-07", "endDate": "2016-06-10"}
  ]
}
