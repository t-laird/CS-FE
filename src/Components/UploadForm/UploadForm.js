import React, { Component } from 'react';
import './UploadForm.css';

class UploadForm extends Component {
  constructor() {
    super();
    this.state = {
      uploadMsg: null,
      uploadSuccessful: true,
      showUpload: false
    };
  }

  badUpload = (prop) => {
    this.setState({ uploadSuccessful: false, uploadMsg: `Invalid JSON file - Missing ${prop} data.` });
    setTimeout(() => { this.setState({ uploadMsg: null}) }, 2000);
  }

  invalidUpload = () => {
    this.setState({ uploadSuccessful: false, uploadMsg: `Please upload a valid JSON file` });
    setTimeout(() => { this.setState({ uploadMsg: null}) }, 2000);
  }

  successfulUpload = () => {
    this.setState({ uploadSuccessful: true, uploadMsg: `Successfully uploaded file.` });
    
    setTimeout(() => { this.setState({ showUpload: false, uploadMsg: null}) }, 2000);
  }

  handleUpload = (e) => {
    e.preventDefault();
    const file = this.fileUpload.files;
    
    if (!file.length) return;

    const fileReader = new FileReader();

    fileReader.onload = (e) => {
      try {
        const result = JSON.parse(e.target.result);
        const expectedProps = ['search', 'campsites', 'reservations'];
        for (const prop in expectedProps) {
          if (!result[expectedProps[prop]]) {
            return this.badUpload(expectedProps[prop]);
          }
        }
        this.props.updateData(result);
        this.successfulUpload();
       
      } catch (err) {
        return this.invalidUpload()
      }
    }

    fileReader.readAsText(file.item(0));
  }

  render() {
    const statusColor = this.state.uploadSuccessful ? 'pos' : 'neg';
    
    return (
      <div className="UploadForm">
      { !this.state.showUpload && 
        <button className="showUpload" onClick={() => {this.setState({showUpload: true})}}>Upload File</button>
      }
      { this.state.showUpload && 
        <form>
          <input type="file" ref={(input) => this.fileUpload = input} />
          <button onClick={this.handleUpload}>Upload</button>
          <span className={statusColor}>{this.state.uploadMsg}</span>
        </form>
      }
      </div>
    );
  }
}
export default UploadForm;