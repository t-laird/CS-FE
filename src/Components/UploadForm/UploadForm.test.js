import React from 'react';
import { mount } from 'enzyme';
import UploadForm from './UploadForm';
import mockData from './mockData';

describe('Upload form tests', () => {
  let renderedForm;
  beforeEach(() => {
    renderedForm = mount(<UploadForm />);
  });

  it('Should match the snapshot', () => {
    expect(renderedForm).toMatchSnapshot();
  });

  it('Should render the correct default components', () => {
    expect(renderedForm.find('UploadForm').length).toEqual(1);
    expect(renderedForm.find('.showUpload').length).toEqual(1);
    expect(renderedForm.find('form').length).toEqual(0);
  });

  it('Should show the form after clicking the button', () => {
    expect(renderedForm.state().showUpload).toEqual(false);
    
    renderedForm.find('.showUpload').simulate('click');

    expect(renderedForm.find('.showUpload').length).toEqual(0);
    expect(renderedForm.find('form').length).toEqual(1);
    expect(renderedForm.state().showUpload).toEqual(true);
  });

  it('Should have the right default state', () => {
    const defaultState = {
      uploadMsg: null,
      uploadSuccessful: true,
      showUpload: false
    };

    expect(renderedForm.state()).toEqual(defaultState);
  });

  it('Should set state correctly given an invalid upload type', () => {
    const expectedState = {
      "showUpload": false,
      "uploadMsg": "Please upload a valid JSON file",
      "uploadSuccessful": false
    };
    renderedForm.instance().invalidUpload();
    expect(renderedForm.state()).toEqual(expectedState);
  });

  it('Should set state correctly given a JSON upload with incomplete information', () => {
    const expectedState = {
      "showUpload": false,
      "uploadMsg": "Invalid JSON file - Missing search data.",
      "uploadSuccessful": false
    };
    renderedForm.instance().badUpload('search');
    expect(renderedForm.state()).toEqual(expectedState);
  });


});