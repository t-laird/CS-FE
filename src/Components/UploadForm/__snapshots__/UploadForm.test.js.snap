// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload form tests Should match the snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <UploadForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": UploadForm {
      "_reactInternalFiber": FiberNode {
        "_debugID": 4,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": FiberNode {
          "_debugID": 3,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": null,
          "_debugSource": null,
          "alternate": null,
          "child": [Circular],
          "effectTag": 1,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "internalContextTag": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "Component": [Function],
            "context": null,
            "props": Object {},
          },
          "memoizedState": Object {
            "context": null,
            "mount": true,
            "props": Object {},
          },
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": FiberNode {
            "_debugID": 1,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": null,
            "_debugSource": null,
            "alternate": FiberNode {
              "_debugID": 1,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": null,
              "_debugSource": null,
              "alternate": [Circular],
              "child": null,
              "effectTag": 0,
              "expirationTime": 1,
              "firstEffect": null,
              "index": 0,
              "internalContextTag": 0,
              "key": null,
              "lastEffect": null,
              "memoizedProps": null,
              "memoizedState": null,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <div
                                    class="UploadForm"
                  >
                                    <button
                                                      class="showUpload"
                                    >
                                                      Upload File
                                    </button>
                  </div>
</div>,
                "context": Object {},
                "current": [Circular],
                "finishedWork": null,
                "hydrate": false,
                "isReadyForCommit": false,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingContext": null,
                "remainingExpirationTime": 0,
              },
              "tag": 3,
              "type": null,
              "updateQueue": Object {
                "baseState": null,
                "callbackList": null,
                "expirationTime": 1,
                "first": Object {
                  "callback": null,
                  "expirationTime": 1,
                  "isForced": false,
                  "isReplace": false,
                  "next": null,
                  "nextCallback": null,
                  "partialState": Object {
                    "element": <WrapperComponent
                      Component={[Function]}
                      context={null}
                      props={Object {}}
/>,
                  },
                },
                "hasForceUpdate": false,
                "isInitialized": false,
                "isProcessing": false,
                "last": Object {
                  "callback": null,
                  "expirationTime": 1,
                  "isForced": false,
                  "isReplace": false,
                  "next": null,
                  "nextCallback": null,
                  "partialState": Object {
                    "element": <WrapperComponent
                      Component={[Function]}
                      context={null}
                      props={Object {}}
/>,
                  },
                },
              },
            },
            "child": [Circular],
            "effectTag": 0,
            "expirationTime": 0,
            "firstEffect": [Circular],
            "index": 0,
            "internalContextTag": 0,
            "key": null,
            "lastEffect": [Circular],
            "memoizedProps": null,
            "memoizedState": Object {
              "element": <WrapperComponent
                Component={[Function]}
                context={null}
                props={Object {}}
/>,
            },
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": null,
            "sibling": null,
            "stateNode": Object {
              "containerInfo": <div>
                <div
                                class="UploadForm"
                >
                                <button
                                                class="showUpload"
                                >
                                                Upload File
                                </button>
                </div>
</div>,
              "context": Object {},
              "current": [Circular],
              "finishedWork": null,
              "hydrate": false,
              "isReadyForCommit": false,
              "nextScheduledRoot": null,
              "pendingChildren": null,
              "pendingContext": null,
              "remainingExpirationTime": 0,
            },
            "tag": 3,
            "type": null,
            "updateQueue": null,
          },
          "sibling": null,
          "stateNode": WrapperComponent {
            "_reactInternalFiber": [Circular],
            "_reactInternalInstance": Object {},
            "context": Object {},
            "props": Object {
              "Component": [Function],
              "context": null,
              "props": Object {},
            },
            "refs": Object {},
            "state": Object {
              "context": null,
              "mount": true,
              "props": Object {},
            },
            "updater": Object {
              "enqueueForceUpdate": [Function],
              "enqueueReplaceState": [Function],
              "enqueueSetState": [Function],
              "isMounted": [Function],
            },
          },
          "tag": 2,
          "type": [Function],
          "updateQueue": null,
        },
        "_debugSource": null,
        "alternate": null,
        "child": FiberNode {
          "_debugID": 5,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": [Circular],
          "_debugSource": Object {
            "fileName": "/Users/thomaslaird/Turing/mod 5/campspot_challenge/src/Components/UploadForm/UploadForm.js",
            "lineNumber": 62,
          },
          "alternate": null,
          "child": FiberNode {
            "_debugID": 6,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": [Circular],
            "_debugSource": Object {
              "fileName": "/Users/thomaslaird/Turing/mod 5/campspot_challenge/src/Components/UploadForm/UploadForm.js",
              "lineNumber": 64,
            },
            "alternate": null,
            "child": null,
            "effectTag": 0,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "internalContextTag": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "children": "Upload File",
              "className": "showUpload",
              "onClick": [Function],
            },
            "memoizedState": null,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": [Circular],
            "sibling": null,
            "stateNode": <button
              class="showUpload"
>
              Upload File
</button>,
            "tag": 5,
            "type": "button",
            "updateQueue": null,
          },
          "effectTag": 0,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "internalContextTag": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "children": Array [
              <button
                className="showUpload"
                onClick={[Function]}
>
                Upload File
</button>,
              false,
            ],
            "className": "UploadForm",
          },
          "memoizedState": null,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": [Circular],
          "sibling": null,
          "stateNode": <div
            class="UploadForm"
>
            <button
                        class="showUpload"
            >
                        Upload File
            </button>
</div>,
          "tag": 5,
          "type": "div",
          "updateQueue": null,
        },
        "effectTag": 1,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "internalContextTag": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {},
        "memoizedState": Object {
          "showUpload": false,
          "uploadMsg": null,
          "uploadSuccessful": true,
        },
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": FiberNode {
          "_debugID": 3,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": null,
          "_debugSource": null,
          "alternate": null,
          "child": [Circular],
          "effectTag": 1,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "internalContextTag": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "Component": [Function],
            "context": null,
            "props": Object {},
          },
          "memoizedState": Object {
            "context": null,
            "mount": true,
            "props": Object {},
          },
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": FiberNode {
            "_debugID": 1,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": null,
            "_debugSource": null,
            "alternate": FiberNode {
              "_debugID": 1,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": null,
              "_debugSource": null,
              "alternate": [Circular],
              "child": null,
              "effectTag": 0,
              "expirationTime": 1,
              "firstEffect": null,
              "index": 0,
              "internalContextTag": 0,
              "key": null,
              "lastEffect": null,
              "memoizedProps": null,
              "memoizedState": null,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <div
                                    class="UploadForm"
                  >
                                    <button
                                                      class="showUpload"
                                    >
                                                      Upload File
                                    </button>
                  </div>
</div>,
                "context": Object {},
                "current": [Circular],
                "finishedWork": null,
                "hydrate": false,
                "isReadyForCommit": false,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingContext": null,
                "remainingExpirationTime": 0,
              },
              "tag": 3,
              "type": null,
              "updateQueue": Object {
                "baseState": null,
                "callbackList": null,
                "expirationTime": 1,
                "first": Object {
                  "callback": null,
                  "expirationTime": 1,
                  "isForced": false,
                  "isReplace": false,
                  "next": null,
                  "nextCallback": null,
                  "partialState": Object {
                    "element": <WrapperComponent
                      Component={[Function]}
                      context={null}
                      props={Object {}}
/>,
                  },
                },
                "hasForceUpdate": false,
                "isInitialized": false,
                "isProcessing": false,
                "last": Object {
                  "callback": null,
                  "expirationTime": 1,
                  "isForced": false,
                  "isReplace": false,
                  "next": null,
                  "nextCallback": null,
                  "partialState": Object {
                    "element": <WrapperComponent
                      Component={[Function]}
                      context={null}
                      props={Object {}}
/>,
                  },
                },
              },
            },
            "child": [Circular],
            "effectTag": 0,
            "expirationTime": 0,
            "firstEffect": [Circular],
            "index": 0,
            "internalContextTag": 0,
            "key": null,
            "lastEffect": [Circular],
            "memoizedProps": null,
            "memoizedState": Object {
              "element": <WrapperComponent
                Component={[Function]}
                context={null}
                props={Object {}}
/>,
            },
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": null,
            "sibling": null,
            "stateNode": Object {
              "containerInfo": <div>
                <div
                                class="UploadForm"
                >
                                <button
                                                class="showUpload"
                                >
                                                Upload File
                                </button>
                </div>
</div>,
              "context": Object {},
              "current": [Circular],
              "finishedWork": null,
              "hydrate": false,
              "isReadyForCommit": false,
              "nextScheduledRoot": null,
              "pendingChildren": null,
              "pendingContext": null,
              "remainingExpirationTime": 0,
            },
            "tag": 3,
            "type": null,
            "updateQueue": null,
          },
          "sibling": null,
          "stateNode": WrapperComponent {
            "_reactInternalFiber": [Circular],
            "_reactInternalInstance": Object {},
            "context": Object {},
            "props": Object {
              "Component": [Function],
              "context": null,
              "props": Object {},
            },
            "refs": Object {},
            "state": Object {
              "context": null,
              "mount": true,
              "props": Object {},
            },
            "updater": Object {
              "enqueueForceUpdate": [Function],
              "enqueueReplaceState": [Function],
              "enqueueSetState": [Function],
              "isMounted": [Function],
            },
          },
          "tag": 2,
          "type": [Function],
          "updateQueue": null,
        },
        "sibling": null,
        "stateNode": [Circular],
        "tag": 2,
        "type": [Function],
        "updateQueue": null,
      },
      "_reactInternalInstance": Object {},
      "badUpload": [Function],
      "context": Object {},
      "handleUpload": [Function],
      "invalidUpload": [Function],
      "props": Object {},
      "refs": Object {},
      "state": Object {
        "showUpload": false,
        "uploadMsg": null,
        "uploadSuccessful": true,
      },
      "successfulUpload": [Function],
      "updater": Object {
        "enqueueForceUpdate": [Function],
        "enqueueReplaceState": [Function],
        "enqueueSetState": [Function],
        "isMounted": [Function],
      },
    },
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="UploadForm"
>
        <button
                class="showUpload"
        >
                Upload File
        </button>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="showUpload"
            onClick={[Function]}
>
            Upload File
</button>,
          false,
        ],
        "className": "UploadForm",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <button
            class="showUpload"
>
            Upload File
</button>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Upload File",
            "className": "showUpload",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Upload File",
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": UploadForm {
        "_reactInternalFiber": FiberNode {
          "_debugID": 4,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": FiberNode {
            "_debugID": 3,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": null,
            "_debugSource": null,
            "alternate": null,
            "child": [Circular],
            "effectTag": 1,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "internalContextTag": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "Component": [Function],
              "context": null,
              "props": Object {},
            },
            "memoizedState": Object {
              "context": null,
              "mount": true,
              "props": Object {},
            },
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": FiberNode {
              "_debugID": 1,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": null,
              "_debugSource": null,
              "alternate": FiberNode {
                "_debugID": 1,
                "_debugIsCurrentlyTiming": false,
                "_debugOwner": null,
                "_debugSource": null,
                "alternate": [Circular],
                "child": null,
                "effectTag": 0,
                "expirationTime": 1,
                "firstEffect": null,
                "index": 0,
                "internalContextTag": 0,
                "key": null,
                "lastEffect": null,
                "memoizedProps": null,
                "memoizedState": null,
                "nextEffect": null,
                "pendingProps": null,
                "ref": null,
                "return": null,
                "sibling": null,
                "stateNode": Object {
                  "containerInfo": <div>
                    <div
                                        class="UploadForm"
                    >
                                        <button
                                                            class="showUpload"
                                        >
                                                            Upload File
                                        </button>
                    </div>
</div>,
                  "context": Object {},
                  "current": [Circular],
                  "finishedWork": null,
                  "hydrate": false,
                  "isReadyForCommit": false,
                  "nextScheduledRoot": null,
                  "pendingChildren": null,
                  "pendingContext": null,
                  "remainingExpirationTime": 0,
                },
                "tag": 3,
                "type": null,
                "updateQueue": Object {
                  "baseState": null,
                  "callbackList": null,
                  "expirationTime": 1,
                  "first": Object {
                    "callback": null,
                    "expirationTime": 1,
                    "isForced": false,
                    "isReplace": false,
                    "next": null,
                    "nextCallback": null,
                    "partialState": Object {
                      "element": <WrapperComponent
                        Component={[Function]}
                        context={null}
                        props={Object {}}
/>,
                    },
                  },
                  "hasForceUpdate": false,
                  "isInitialized": false,
                  "isProcessing": false,
                  "last": Object {
                    "callback": null,
                    "expirationTime": 1,
                    "isForced": false,
                    "isReplace": false,
                    "next": null,
                    "nextCallback": null,
                    "partialState": Object {
                      "element": <WrapperComponent
                        Component={[Function]}
                        context={null}
                        props={Object {}}
/>,
                    },
                  },
                },
              },
              "child": [Circular],
              "effectTag": 0,
              "expirationTime": 0,
              "firstEffect": [Circular],
              "index": 0,
              "internalContextTag": 0,
              "key": null,
              "lastEffect": [Circular],
              "memoizedProps": null,
              "memoizedState": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={Object {}}
/>,
              },
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <div
                                    class="UploadForm"
                  >
                                    <button
                                                      class="showUpload"
                                    >
                                                      Upload File
                                    </button>
                  </div>
</div>,
                "context": Object {},
                "current": [Circular],
                "finishedWork": null,
                "hydrate": false,
                "isReadyForCommit": false,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingContext": null,
                "remainingExpirationTime": 0,
              },
              "tag": 3,
              "type": null,
              "updateQueue": null,
            },
            "sibling": null,
            "stateNode": WrapperComponent {
              "_reactInternalFiber": [Circular],
              "_reactInternalInstance": Object {},
              "context": Object {},
              "props": Object {
                "Component": [Function],
                "context": null,
                "props": Object {},
              },
              "refs": Object {},
              "state": Object {
                "context": null,
                "mount": true,
                "props": Object {},
              },
              "updater": Object {
                "enqueueForceUpdate": [Function],
                "enqueueReplaceState": [Function],
                "enqueueSetState": [Function],
                "isMounted": [Function],
              },
            },
            "tag": 2,
            "type": [Function],
            "updateQueue": null,
          },
          "_debugSource": null,
          "alternate": null,
          "child": FiberNode {
            "_debugID": 5,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": [Circular],
            "_debugSource": Object {
              "fileName": "/Users/thomaslaird/Turing/mod 5/campspot_challenge/src/Components/UploadForm/UploadForm.js",
              "lineNumber": 62,
            },
            "alternate": null,
            "child": FiberNode {
              "_debugID": 6,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": [Circular],
              "_debugSource": Object {
                "fileName": "/Users/thomaslaird/Turing/mod 5/campspot_challenge/src/Components/UploadForm/UploadForm.js",
                "lineNumber": 64,
              },
              "alternate": null,
              "child": null,
              "effectTag": 0,
              "expirationTime": 0,
              "firstEffect": null,
              "index": 0,
              "internalContextTag": 0,
              "key": null,
              "lastEffect": null,
              "memoizedProps": Object {
                "children": "Upload File",
                "className": "showUpload",
                "onClick": [Function],
              },
              "memoizedState": null,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": [Circular],
              "sibling": null,
              "stateNode": <button
                class="showUpload"
>
                Upload File
</button>,
              "tag": 5,
              "type": "button",
              "updateQueue": null,
            },
            "effectTag": 0,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "internalContextTag": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "children": Array [
                <button
                  className="showUpload"
                  onClick={[Function]}
>
                  Upload File
</button>,
                false,
              ],
              "className": "UploadForm",
            },
            "memoizedState": null,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": [Circular],
            "sibling": null,
            "stateNode": <div
              class="UploadForm"
>
              <button
                            class="showUpload"
              >
                            Upload File
              </button>
</div>,
            "tag": 5,
            "type": "div",
            "updateQueue": null,
          },
          "effectTag": 1,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "internalContextTag": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {},
          "memoizedState": Object {
            "showUpload": false,
            "uploadMsg": null,
            "uploadSuccessful": true,
          },
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": FiberNode {
            "_debugID": 3,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": null,
            "_debugSource": null,
            "alternate": null,
            "child": [Circular],
            "effectTag": 1,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "internalContextTag": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "Component": [Function],
              "context": null,
              "props": Object {},
            },
            "memoizedState": Object {
              "context": null,
              "mount": true,
              "props": Object {},
            },
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": FiberNode {
              "_debugID": 1,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": null,
              "_debugSource": null,
              "alternate": FiberNode {
                "_debugID": 1,
                "_debugIsCurrentlyTiming": false,
                "_debugOwner": null,
                "_debugSource": null,
                "alternate": [Circular],
                "child": null,
                "effectTag": 0,
                "expirationTime": 1,
                "firstEffect": null,
                "index": 0,
                "internalContextTag": 0,
                "key": null,
                "lastEffect": null,
                "memoizedProps": null,
                "memoizedState": null,
                "nextEffect": null,
                "pendingProps": null,
                "ref": null,
                "return": null,
                "sibling": null,
                "stateNode": Object {
                  "containerInfo": <div>
                    <div
                                        class="UploadForm"
                    >
                                        <button
                                                            class="showUpload"
                                        >
                                                            Upload File
                                        </button>
                    </div>
</div>,
                  "context": Object {},
                  "current": [Circular],
                  "finishedWork": null,
                  "hydrate": false,
                  "isReadyForCommit": false,
                  "nextScheduledRoot": null,
                  "pendingChildren": null,
                  "pendingContext": null,
                  "remainingExpirationTime": 0,
                },
                "tag": 3,
                "type": null,
                "updateQueue": Object {
                  "baseState": null,
                  "callbackList": null,
                  "expirationTime": 1,
                  "first": Object {
                    "callback": null,
                    "expirationTime": 1,
                    "isForced": false,
                    "isReplace": false,
                    "next": null,
                    "nextCallback": null,
                    "partialState": Object {
                      "element": <WrapperComponent
                        Component={[Function]}
                        context={null}
                        props={Object {}}
/>,
                    },
                  },
                  "hasForceUpdate": false,
                  "isInitialized": false,
                  "isProcessing": false,
                  "last": Object {
                    "callback": null,
                    "expirationTime": 1,
                    "isForced": false,
                    "isReplace": false,
                    "next": null,
                    "nextCallback": null,
                    "partialState": Object {
                      "element": <WrapperComponent
                        Component={[Function]}
                        context={null}
                        props={Object {}}
/>,
                    },
                  },
                },
              },
              "child": [Circular],
              "effectTag": 0,
              "expirationTime": 0,
              "firstEffect": [Circular],
              "index": 0,
              "internalContextTag": 0,
              "key": null,
              "lastEffect": [Circular],
              "memoizedProps": null,
              "memoizedState": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={Object {}}
/>,
              },
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <div
                                    class="UploadForm"
                  >
                                    <button
                                                      class="showUpload"
                                    >
                                                      Upload File
                                    </button>
                  </div>
</div>,
                "context": Object {},
                "current": [Circular],
                "finishedWork": null,
                "hydrate": false,
                "isReadyForCommit": false,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingContext": null,
                "remainingExpirationTime": 0,
              },
              "tag": 3,
              "type": null,
              "updateQueue": null,
            },
            "sibling": null,
            "stateNode": WrapperComponent {
              "_reactInternalFiber": [Circular],
              "_reactInternalInstance": Object {},
              "context": Object {},
              "props": Object {
                "Component": [Function],
                "context": null,
                "props": Object {},
              },
              "refs": Object {},
              "state": Object {
                "context": null,
                "mount": true,
                "props": Object {},
              },
              "updater": Object {
                "enqueueForceUpdate": [Function],
                "enqueueReplaceState": [Function],
                "enqueueSetState": [Function],
                "isMounted": [Function],
              },
            },
            "tag": 2,
            "type": [Function],
            "updateQueue": null,
          },
          "sibling": null,
          "stateNode": [Circular],
          "tag": 2,
          "type": [Function],
          "updateQueue": null,
        },
        "_reactInternalInstance": Object {},
        "badUpload": [Function],
        "context": Object {},
        "handleUpload": [Function],
        "invalidUpload": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "showUpload": false,
          "uploadMsg": null,
          "uploadSuccessful": true,
        },
        "successfulUpload": [Function],
        "updater": Object {
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
        },
      },
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="UploadForm"
>
          <button
                    class="showUpload"
          >
                    Upload File
          </button>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="showUpload"
              onClick={[Function]}
>
              Upload File
</button>,
            false,
          ],
          "className": "UploadForm",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <button
              class="showUpload"
>
              Upload File
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Upload File",
              "className": "showUpload",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Upload File",
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
